epix/run_epix.py

```diff
@@ -0,0 +1,202 @@
+import os
+import time
+import awkward as ak
+import numpy as np
+import pandas as pd
+import warnings
+
+import epix
+
+
+def main(args, return_df=False, return_wfsim_instructions=False, strax=False):
+    """Call this function from the run_epix script"""
+
+    if args['debug']:
+        print("epix configuration: ", args)
+        # TODO: also add memory information see starxer and change this to debug
+        # Getting time information:
+        starttime = time.time()
+        tnow = starttime
+
+    # Loading data:
+    inter, n_simulated_events = epix.loader(args['path'],
+                                            args['file_name'],
+                                            args['debug'],
+                                            outer_cylinder=args['outer_cylinder'],
+                                            kwargs_uproot_arrays={'entry_start': args['entry_start'],
+                                                                  'entry_stop': args['entry_stop']}
+                                            )
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'load data.')
+        print(f"Finding clusters of interactions with a dr = {args['micro_separation']} mm"
+               f" and dt = {args['micro_separation_time']} ns")
+
+    # Cluster finding and clustering:
+    inter = epix.find_cluster(inter, args['micro_separation']/10, args['micro_separation_time'])
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'cluster finding.')
+
+    result = epix.cluster(inter, args['tag_cluster_by'] == 'energy')
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'cluster merging.')
+
+    # Add eventid again:
+    result['evtid'] = ak.broadcast_arrays(inter['evtid'][:, 0], result['ed'])[0]
+
+    # Sort detector volumes and keep interactions in selected ones:
+    if args['debug']:
+        print('Removing clusters not in volumes:', *[v.name for v in args['detector_config']])
+        print(f'Number of clusters before: {np.sum(ak.num(result["ed"]))}')
+
+    # Returns all interactions which are inside in one of the volumes,
+    # Checks for volume overlap, assigns Xe density and create S2 to
+    # interactions. EField comes later since interpolated maps cannot be
+    # called inside numba functions.
+    res_det = epix.in_sensitive_volume(result, args['detector_config'])
+
+    # Adding new fields to result:
+    for field in res_det.fields:
+        result[field] = res_det[field]
+    m = result['vol_id'] > 0  # All volumes have an id larger zero
+    result = result[m]
+
+    # Removing now empty events as a result of the selection above:
+    m = ak.num(result['ed']) > 0
+    result = result[m]
```

Comment:
These cuts should not have an impact on the accurate source rate - event time calculations or?

Reply:
No, this is why I added n_events_simulated, to get the correct source rate.